sql/inspect: exclude REFCURSOR[] from index consistency comparisons

spilchen opened the pull request:
The index consistency check previously filtered out REFCURSOR columns when constructing join predicates, but failed to do the same for REFCURSOR[] (arrays of cursors). Like REFCURSOR, these cannot be used in comparison operations.

This change ensures that both REFCURSOR and REFCURSOR[] columns are properly excluded from such comparisons.

Informs: #155483
Epic: CRDB-55075
Release note: none

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

One suggestion above to replace the existing check and make it more future proof.

> @fqazi reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156273#-:-OcaDWfLE3EjnxSy2zSV:bxwn8h5)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/sql/inspect/index_consistency_check.go` line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/156273#-OcaD8Q726i6P12KxeFq:-OcaD8Q726i6P12KxeFr:bz4pnev) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d0647a371598ba32e2f25a11a007b69e555c242/pkg/sql/inspect/index_consistency_check.go#L152)):*
> ```Go
> 		col := c.tableDesc.PublicColumns()[pos]
> 		otherColumns = append(otherColumns, col)
> 		if isRefCursorType(col.GetType()) {
> ```

Should we just use `tree.EqCmpAllowedForEquivalentTypes` instead? That will also make it more future proof.


<!-- Sent from Reviewable.io -->

spilchen committed:
sql/inspect: exclude REFCURSOR[] from index consistency comparisons

The index consistency check previously filtered out REFCURSOR columns when
constructing join predicates, but failed to do the same for REFCURSOR[] (arrays
of cursors). Like REFCURSOR, these cannot be used in comparison operations.

This change ensures that both REFCURSOR and REFCURSOR[] columns are properly
excluded from such comparisons.

Informs: #155483
Epic: CRDB-55075
Release note: none

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156273)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @fqazi)

___
*[`pkg/sql/inspect/index_consistency_check.go` line 152 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/156273#-OcaD8Q726i6P12KxeFq:-Ocac8WXAGG1S_I7I8aE:bfsnw66) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d0647a371598ba32e2f25a11a007b69e555c242/pkg/sql/inspect/index_consistency_check.go#L152)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Should we just use `tree.EqCmpAllowedForEquivalentTypes` instead? That will also make it more future proof.
</blockquote></details>

Good idea. This made it a lot cleaner!


<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
Nice!

> @fqazi reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156273#-:-OcacfSkBHghGc63LJKz:b-83sjoh)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @spilchen)



<!-- Sent from Reviewable.io -->
